### PR TITLE
sv-plugin: Strip the node name using parent name instead of "."

### PR DESCRIPTION
**Pull request by mandrys**
Considering the fact that the name can consist of escaped characters, including dot, we need to change the way of stripping the name of a node - instead of a dot, we now use the ancestor name. 

Added test: https://github.com/chipsalliance/UHDM-integration-tests/pull/730 (It fails formal verification because of sv2v being unable to handle escaped characters, but is useful as a parsing test.)

Test yosys-systemverilog CI run: https://github.com/antmicro/yosys-systemverilog/actions/runs/4916684728.

### CODE REVIEW

File: systemverilog-plugin/UhdmAst.cc [hunk 1]
```diff
@@ -183,8 +183,75 @@ static std::string get_name(vpiHandle obj_h, bool prefer_full_name = false)
...
+            parent_h = vpi_handle(vpiParent, parent_h);
+            if (parent_h) {
+                if (auto p = vpi_get_str(vpiFullName, parent_h)) {
+                    parent_name = p;
+                } else if (auto p = vpi_get_str(vpiName, parent_h)) {
+                    parent_name = p;
+                } else if (auto p = vpi_get_str(vpiDefName, parent_h)) {
+                    parent_name = p;
+                }
+                while (!parent_name.empty()) {
```
Comment (kamilrakoczy) [inline systemverilog-plugin/UhdmAst.cc:213]: I think this requires some refactoring as there is a lot of code repetition. Please extract recurrent patterns to separate functions.

**Review by kamilrakoczy — CHANGES_REQUESTED** [review]

**Comment by wsipak** [comment]
We've had an offline discussion about the `vpi_handle`s and the other API of UHDM that allows dealing with objects, that we could consider here. I think using the `UHDM::BaseClass` -based stuff could also help us keep the code cleaner.

File: systemverilog-plugin/UhdmAst.cc (hunk 1 above)
Comment (mandrys) [reply systemverilog-plugin/UhdmAst.cc]: Done.

**Comment by mandrys** [comment]
New _yosys-systemverilog_ run: https://github.com/antmicro/yosys-systemverilog/actions/runs/5134240233.

File: systemverilog-plugin/UhdmAst.cc [hunk 2]
```diff
@@ -186,9 +219,30 @@ static std::string get_name(vpiHandle obj_h, bool prefer_full_name = false)
...
+        parent_name = get_parent_name(parent_h);
+
+        if (parent_name.empty()) {
+            // Nodes of certain types, like param_assign, don't have
+            // a name, so we need to look further for the ancestor.
+            parent_h = vpi_handle(vpiParent, parent_h);
+
+            if (parent_h) {
+                parent_name = get_parent_name(parent_h);
+                find_ancestor_name(parent_h, name, parent_name);
```
Comment (mglb) [inline systemverilog-plugin/UhdmAst.cc:240]:
```suggestion
```
I think `find_ancestor_name` call below will handle it. Is this correct?

File: systemverilog-plugin/UhdmAst.cc [hunk 3]
```diff
@@ -186,9 +219,30 @@ static std::string get_name(vpiHandle obj_h, bool prefer_full_name = false)
...
+            }
+        }
+        find_ancestor_name(parent_h, name, parent_name);
     }
+    vpi_release_handle(parent_h);
```
Comment (mglb) [inline systemverilog-plugin/UhdmAst.cc:245]:
```suggestion
```
Handle released in find_ancestor_name above.

**Review by mglb — CHANGES_REQUESTED** [review]
Few changes; please re-run yosys-systemverilog with it. If it passes it will mean I didn't make any mistake.

**Comment by mandrys** [comment]
Latest _yosys-systemverilog_ run passed, only formal verification failed for some unexpected passes: https://github.com/antmicro/yosys-systemverilog/actions/runs/5134720743/jobs/9239550297.
When the PR is approved, I'll squash the commits with fixes and merge it.

**Review by mglb — APPROVED** [review]

**Pull request merged**